### PR TITLE
Feat/16 usernamepw

### PR DESCRIPTION
**Pull request by jdsauer**
### Changes made by this Pull Request

- Show username in web app

### Definition of Done

- [x] Tests are written and run successful
- [x] Code is documented
- [x] User Manual is updated
- [x] All requirements of the user story are met

### CODE REVIEW

**Comment by jb3rndt** [comment]
Is there a specific reason why the test files for component tests are `.js` instead of `.ts` now?

**Comment by jdsauer** [comment]
> Is there a specific reason why the test files for component tests are `.js` instead of `.ts` now?

it fixed my path errors and it doesn't changed the outcome of the tests

File: static/src/components/UserInformationLabel.vue [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  props: {
+    username: { type: String, required: true },
+  },
+});
+</script>
+
+<template>
+  <div>
+    Current logged in user:
+    <label id="userLabel" for="username">{{ username }}</label>
```
Comment (bewee) [inline static/src/components/UserInformationLabel.vue]: From my understanding, the `for="username"` is only used to create a relation to an input field. So is this really of any use here?

File: static/src/components/UserInformationLabel.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  </div>
+</template>
+
+<style>
+label[for="username"] {
```
Comment (bewee) [inline static/src/components/UserInformationLabel.vue]: (Same here)

File: static/src/components/__tests__/UserInformationLabel.spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+describe("UserInformationLabel", () => {
+  it("exists", () => {
+    mount(UserInformationLabel, { props: { username: "" } });
+  });
+});
+
+// Label renders correctly and is visible
+describe("App", () => {
+  it("renders properly", () => {
+    mount(App, { probs: { username: "" } });
```
Comment (bewee) [inline static/src/components/__tests__/UserInformationLabel.spec.js]:
```suggestion
    mount(UserInformationLabel, { props: { username: "" } });
```
This test seems a little fishy to me, I think you rather wanted to do this?

File: static/src/components/__tests__/UserInformationLabel.spec.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    cy.get("#userLabel").should("be.visible");
+  });
+});
+
+// Label renders correctly and is visible
+describe("App", () => {
+  it("shows username", () => {
+    mount(App, { probs: { username: "" } });
```
Comment (bewee) [inline static/src/components/__tests__/UserInformationLabel.spec.js]:
```suggestion
    mount(UserInformationLabel, { props: { username: "" } });
```

**Review by bewee — COMMENTED** [review]
Good work!
I'd suggest renaming the test files back to `.ts` so that we don't loose static type checks. You could however try to change some vscode settings so that this will no longer be falsely marked in the IDE.

**Review by bewee — APPROVED** [review]

**Review by n-koenig — APPROVED** [review]

**Pull request merged**